pandas/core/ops/array_ops.py

```diff
@@ -293,11 +292,10 @@ def na_logical_op(x: np.ndarray, y, op):
                 OverflowError,
                 NotImplementedError,
             ):
+                typ = type(y).__name__
                 raise TypeError(
-                    "Cannot perform '{op}' with a dtyped [{dtype}] array "
-                    "and scalar of type [{typ}]".format(
-                        op=op.__name__, dtype=x.dtype, typ=type(y).__name__
-                    )
+                    f"Cannot perform '{op.__name__}' with a dtyped [{x.dtype}] array "
```

Comment:
i agree with @TomAugspurger here - i would simply in-line this rather than creating a temporary

temporary only if it’s a really complicated expr or has quoting itself (eg a .join(..)